modules/v4l2/v4l2.c

```diff
@@ -524,14 +547,30 @@ static int alloc(struct vidsrc_st **stp, const struct vidsrc *vs,
 
 static int v4l_init(void)
 {
-	return vidsrc_register(&vidsrc, baresip_vidsrcl(),
+	int err;
+
+	err = vidsrc_register(&vidsrc, baresip_vidsrcl(),
 			       "v4l2", alloc, NULL);
+
+	list_init(&vidsrc->dev_list);
+	set_available_devices(vidsrc);
+
+	return err;
 }
 
 
 static int v4l_close(void)
 {
+	struct le *le;
+	struct mediadev *dev;
+
+	for (le = list_head(&vidsrc->dev_list); le; le = le->next) {
+		dev = le->data;
+		mem_deref((char*)dev->name);
+	}
+	list_flush(&vidsrc->dev_list);
```

Comment:
perhaps we should move the `list_flush` to the vidsrc destructor ? then there is no need
to add any extra code in the close functions.

vidsrc.c:

```
static void destructor(void *arg)
{
	struct vidsrc *vs = arg;

	list_unlink(&vs->le);
}
```